adal/src/main/java/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -382,6 +382,30 @@ private static Calendar getTokenValidityTime() {
         return timeAhead;
     }
 
+    private boolean validTokenCacheItem(String item){
+        if (item == null) {
+            Logger.w(TAG, "Bad input, was null. ");
+            return false;
+        }
+
+        item = item.trim();
+        if(item.isEmpty()) {
+            Logger.e(TAG, "Bad input, was empty string. ");
+            return false;
+        } else if(!item.contains(":")) {
+            Logger.e(TAG, "Bad input, doesn't contain key pairs. ");
+            return false;
+        } else if(item.charAt(0) != '{') {
+            Logger.e(TAG, "Bad input, start of input is invalid. Expected opening bracket '{'. ");
+            return false;
+        } else if(item.charAt(item.length()-1) != '}') {
```

Comment:
Are these checks enough to validate the json

Reply:
It's trying to validate the tokenCacheItem for being malformed JSON rather than 'validating JSON'. This seemed all we can do for now to avoid PII risk.